scripts/exglobal_atmos_chgres_gen_control.sh

```diff
@@ -0,0 +1,122 @@
+  cpfs "${SOURCE_DIR}/sfc/${CASE}.mx${OCNRES}.slope_type.tile${i}.nc" "${DATA}/"
+  cpfs "${SOURCE_DIR}/sfc/${CASE}.mx${OCNRES}.maximum_snow_albedo.tile${i}.nc" "${DATA}/"
+  cpfs "${SOURCE_DIR}/sfc/${CASE}.mx${OCNRES}.snowfree_albedo.tile${i}.nc" "${DATA}/"
+  cpfs "${SOURCE_DIR}/sfc/${CASE}.mx${OCNRES}.soil_type.tile${i}.nc" "${DATA}/"
+  cpfs "${SOURCE_DIR}/sfc/${CASE}.mx${OCNRES}.vegetation_type.tile${i}.nc" "${DATA}/"
+  cpfs "${SOURCE_DIR}/sfc/${CASE}.mx${OCNRES}.substrate_temperature.tile${i}.nc" "${DATA}/"
+  cpfs "${SOURCE_DIR}/sfc/${CASE}.mx${OCNRES}.vegetation_greenness.tile${i}.nc" "${DATA}/"
+  cpfs "${SOURCE_DIR}/sfc/${CASE}.mx${OCNRES}.facsf.tile${i}.nc" "${DATA}/"
+done
+################################################################################
+# add the namelist and run chgres
+cat << EOF > ./fort.41
+&config
+mosaic_file_target_grid="./${CASE}_mosaic.nc"
+fix_dir_target_grid="./"
+orog_dir_target_grid="./"
+orog_files_target_grid="${CASE}.mx${OCNRES}_oro_data.tile1.nc","${CASE}.mx${OCNRES}_oro_data.tile2.nc","${CASE}.mx${OCNRES}_oro_data.tile3.nc","${CASE}.mx${OCNRES}_oro_data.tile4.nc","${CASE}.mx${OCNRES}_oro_data.tile5.nc","${CASE}.mx${OCNRES}_oro_data.tile6.nc"
+vcoord_file_target_grid="./global_hyblev.l${LEVS}.txt"
+mosaic_file_input_grid="NULL"
+orog_dir_input_grid="NULL"
+orog_files_input_grid="NULL"
+data_dir_input_grid="./"
+atm_files_input_grid="${ATM_FILE}"
+atm_core_files_input_grid="NULL"
+atm_tracer_files_input_grid="NULL"
+sfc_files_input_grid="${SFC_FILE}"
+nst_files_input_grid="NULL"
+grib2_file_input_grid="NULL"
+geogrid_file_input_grid="NULL"
+varmap_file="NULL"
+wam_parm_file="NULL"
+cycle_year=${iy}
+cycle_mon=${im}
+cycle_day=${id}
+cycle_hour=${cyc}
+convert_atm=.true.
+convert_sfc=.true.
+convert_nst=.true.
+input_type="gaussian_netcdf"
+tracers="sphum","liq_wat","o3mr","ice_wat","rainwat","snowwat","graupel"
+tracers_input="spfh","clwmr","o3mr","icmr","rwmr","snmr","grle"
+regional=0
+halo_bndy=0
+halo_blend=0
+sotyp_from_climo=.true.
+vgtyp_from_climo=.true.
+vgfrc_from_climo=.true.
+minmax_vgfrc_from_climo=.true.
+tg3_from_soil=.false.
+lai_from_climo=true.
+external_model="GFS"
+nsoill_out=4
+thomp_mp_climo_file="NULL"
+wam_cold_start=.false.
+/
+EOF
+
+eval "${APRUN_CHGRES}" "${CHGRESEXEC}"
+################################################################################
+# Ensure COMIN_ATMOS_INPUT_MEM exists, create if needed, then copy out.atm.tile{1..6}.nc (force overwrite)
+if [[ ! -d "${COMOUT_ATMOS_INPUT_MEM}" ]]; then
+  if ! mkdir -p "${COMOUT_ATMOS_INPUT_MEM}"; then
+    echo "ERROR: Failed to create directory ${COMOUT_ATMOS_INPUT_MEM}"
+    exit 1
+  fi
+fi
+
+for i in {1..6}; do
+  atm_file="out.atm.tile${i}.nc"
+  sfc_file="out.sfc.tile${i}.nc"
+  if [[ -f "${atm_file}" ]]; then
+    cpfs "${atm_file}" "${COMOUT_ATMOS_INPUT_MEM}/"
+  fi
+  if [[ -f "${sfc_file}" ]]; then
+    cpfs "${sfc_file}" "${COMOUT_ATMOS_INPUT_MEM}/"
+  fi
```

Comment:
```suggestion
  cpreq "${atm_file}" "${COMOUT_ATMOS_INPUT_MEM}/gfs_data.tile${i}.nc"
  cpreq "${sfc_file}" "${COMOUT_ATMOS_INPUT_MEM}/sfc_data.tile${i}.nc"
```
`cpreq` will ensure the file is created and correct.
destination files are called `gfs_data` and `sfc_data` respectively (IIRC)